dioptas/tests/functional_tests/test_mask_save_and_load.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+# Dioptas - GUI program for fast processing of 2D X-ray diffraction data
+# Principal author: Clemens Prescher (clemens.prescher@gmail.com)
+# Copyright (C) 2014-2019 GSECARS, University of Chicago, USA
+# Copyright (C) 2015-2018 Institute for Geology and Mineralogy, University of Cologne, Germany
+# Copyright (C) 2019-2020 DESY, Hamburg, Germany
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os.path
+from unittest.mock import MagicMock
+
+import fabio
+import numpy as np
+import pytest
+from PIL import Image
+from qtpy import QtCore, QtWidgets
+from qtpy.QtTest import QTest
+
+from ..utility import click_button, unittest_data_path, QtTest
+from ...controller.MainController import MainController
+
+
```

Comment:
I made this new test with pytest fixtures. If you prefer that it's use `QtTest`, I'll change this.